govcd/metadata_openapi.go

```diff
@@ -0,0 +1,252 @@
+/*
+ * Copyright 2022 VMware, Inc.  All rights reserved.  Licensed under the Apache v2 License.
+ */
+
+package govcd
+
+import (
+	"fmt"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+	"net/url"
+)
+
+// OpenApiMetadataEntry is a wrapper object for types.OpenApiMetadataEntry
+type OpenApiMetadataEntry struct {
+	MetadataEntry  *types.OpenApiMetadataEntry
+	client         *Client
+	Etag           string // Allows concurrent operations with metadata
+	href           string // This is the HREF of the given metadata entry
+	parentEndpoint string // This is the endpoint of the object that has the metadata entries
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+// Specific objects compatible with metadata
+// ---------------------------------------------------------------------------------------------------------------------
+
+// GetMetadata returns all the metadata from a DefinedEntity.
+// NOTE: The obtained metadata doesn't have ETags, use GetMetadataById or GetMetadataByKey to obtain a ETag for a specific entry.
+func (rde *DefinedEntity) GetMetadata() ([]*OpenApiMetadataEntry, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointRdeEntities
+	return getAllOpenApiMetadata(rde.client, endpoint, rde.DefinedEntity.ID, nil)
+}
+
+// GetMetadataByKey returns a unique DefinedEntity metadata entry corresponding to the given domain, namespace and key.
+// The domain and namespace are only needed when there's more than one entry with the same key.
+// This is a more costly operation than GetMetadataById due to ETags, so use that preferred option whenever possible.
+func (rde *DefinedEntity) GetMetadataByKey(domain, namespace, key string) (*OpenApiMetadataEntry, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointRdeEntities
+	return getOpenApiMetadataByKey(rde.client, endpoint, rde.DefinedEntity.ID, domain, namespace, key)
+}
+
+// GetMetadataById returns a unique DefinedEntity metadata entry corresponding to the given domain, namespace and key.
+// The domain and namespace are only needed when there's more than one entry with the same key.
+func (rde *DefinedEntity) GetMetadataById(id string) (*OpenApiMetadataEntry, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointRdeEntities
+	return getOpenApiMetadataById(rde.client, endpoint, rde.DefinedEntity.ID, id)
+}
+
+// AddMetadata adds metadata to the receiver DefinedEntity.
+func (rde *DefinedEntity) AddMetadata(metadataEntry types.OpenApiMetadataEntry) (*OpenApiMetadataEntry, error) {
+	endpoint := types.OpenApiPathVersion1_0_0 + types.OpenApiEndpointRdeEntities
+	return addOpenApiMetadata(rde.client, endpoint, rde.DefinedEntity.ID, metadataEntry)
+}
+
+// ---------------------------------------------------------------------------------------------------------------------
+// Generic methods for OpenAPI metadata
```

Comment:
```suggestion
// Entity-agnostic methods for OpenAPI metadata
```
OR
```suggestion
// Metadata Entry methods for OpenAPI metadata
```
I think "Generic" gives a false impression to the reader, since Go 1.18

Reply:
Changed, thanks